### DISCUSSION TITLE
Mon onduleur SOFAR HYD 5000 EP ne se connecte pas

### DISCUSSION THREAD

**Discussion by alprech** [davidrapan/ha-solarman#260] [Q&A]
Bonjour David,
En essayant d'intégrer mon onduleur dans HA avec votre solution, j'ai 2 types de messages d'erreur (cf. PJ copie d'écran). Le logger a un numéro de série = 2382921044 et l'IP fixe est 192.168.1.23. J'utilise la configuration sofar_hyd-es.yaml qui est la seule qui me donne une réponse.
Le problème vient il du suffixe EP ou ES. 
![Capture d’écran (17)](https://github.com/user-attachments/assets/b77d835b-f1f2-4987-9a34-9b8395e29a67)
![Capture d’écran (16)](https://github.com/user-attachments/assets/26e029a5-3a59-4006-893f-050a283a2181)
Merci pour votre aide

**Comment by davidrapan** [comment]
Bonjour, cela peut être dû à plusieurs raisons, dont l'une est un profil incorrect (et nécessiterait un nouveau) ou même #172

Pouvez-vous s'il vous plaît activer la journalisation de débogage, puis essayer d'ajouter l'appareil et de joindre le fichier ici par la suite.

**Reply by alprech** [reply]
Bonjour David, merci pour votre réponse ultra rapide et votre suggestion. 
Après avoir écrit ma demande, j'ai essayé "par hasard" le fichier sofar_tlx-g3.yaml qui s'adresse normalement à un onduleur triphasé. MIRACLE (de mon point de vue) : ça fonctionne car la connexion s'établi et remonte 38 entités. Quand elles ne sont pas à zéro, les entités fournissent des valeurs correctes en comparaison avec l'affichage de l'onduleur. 
Bien évidemment, il y a des entités pour les autres phases et rien pour la batterie, ce qui est normal.

Afin de ne pas perdre l'acquis, je vais créer un autre appareil avec le fichier de paramètres sofar_g3hyd.yaml et activer la journalisation de débogage. A bientôt

**Reply by alprech** [reply]
correction : sofar_g3hyd.yaml et non canapér_g3hyd.yaml

**Reply by alprech** [reply]
Suite de mes essais : 
1 - avec le fichier SOFAR_g3hyd.yaml, c'est OK et remonte 137 entités (!) dont celles de la batterie (que je dois vérifier).
2 - avec le fichier SOFAR_hyd-es.yaml, c'est KO et je joins le journal de débogage.
merci pour l'analyse de ce qui coince avec celui qui semblait être le mieux adapté.
UN GRAND MERCI pour la fin de ma galère qui dure depuis quelques semaines.
André
[home-assistant.log](https://github.com/user-attachments/files/17748518/home-assistant.log)

**Reply by davidrapan** [reply]
1. Les valeurs correspondent-elles ?
2. Les registres 0512 - 0597 | 0x0200 - 0x0255 posent problème, essayez de les supprimer du fichier.

**Reply by alprech** [reply]
Bonjour David,
1 - Les valeurs remontées à l'aide du fichier  SOFAR_g3hyd.yaml sont toutes correctes sauf Fault4 qui a raccordé une bétise pendant un jour ou deux mais tout est rentré dans l'ordre. Bien évidemment j'ai trop de capteurs et j'ai supprimé dans HA tout ce qui correspondait aux phases S et T.
2 - Les registres mentionnés ne figurent pas dans le fichier utilisé mais sont présents pour l'onduleur hybride monophasé. Pour supprimer ces registres, faut il supprimer tout ce qui s'y rapporte? Si oui, je ferai un essai bien qu'aujourd'hui ça fonctionne avec le 1.
